std/format.d

```diff
@@ -3656,15 +3656,15 @@ private template hasToString(T, Char)
     {
         enum hasToString = 5;
     }
-    else static if (is(typeof({ T val = void; FormatSpec!Char f; val.toString((const(char)[] s){}, f); })))
+    else static if (is(typeof({ T val = void; FormatSpec!Char f; val.toString((scope const(char)[] s){}, f); })))
     {
         enum hasToString = 4;
     }
-    else static if (is(typeof({ T val = void; val.toString((const(char)[] s){}, "%s"); })))
+    else static if (is(typeof({ T val = void; val.toString((scope const(char)[] s){}, "%s"); })))
     {
         enum hasToString = 3;
     }
-    else static if (is(typeof({ T val = void; val.toString((const(char)[] s){}); })))
+    else static if (is(typeof({ T val = void; val.toString((scope const(char)[] s){}); })))
     {
         enum hasToString = 2;
```

Comment:
I'm not sure whether we can modify these overload checks without a deprecation period.

Reply:
Because of contravariance, it should still work with the non-scope versions.